pkg/operator/controller/externaldns/rbac_test.go

```diff
@@ -0,0 +1,625 @@
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     "otherrole",
+				},
+				Subjects: []rbacv1.Subject{
+					{
+						Kind:      "ServiceAccount",
+						Name:      controller.ExternalDNSResourceName(testExternalDNS),
+						Namespace: testNamespace,
+					},
+				},
+			},
+			inputExpected: &rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: controller.ExternalDNSResourceName(testExternalDNS),
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     controller.ExternalDNSBaseName,
+				},
+				Subjects: []rbacv1.Subject{
+					{
+						Kind:      "ServiceAccount",
+						Name:      controller.ExternalDNSResourceName(testExternalDNS),
+						Namespace: testNamespace,
+					},
+				},
+			},
+			expectChanged: true,
+		},
+		{
+			name: "Subject name changed",
+			inputCurrent: &rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: controller.ExternalDNSResourceName(testExternalDNS),
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     controller.ExternalDNSBaseName,
+				},
+				Subjects: []rbacv1.Subject{
+					{
+						Kind:      "ServiceAccount",
+						Name:      "othersa",
+						Namespace: testNamespace,
+					},
+				},
+			},
+			inputExpected: &rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: controller.ExternalDNSResourceName(testExternalDNS),
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     controller.ExternalDNSBaseName,
+				},
+				Subjects: []rbacv1.Subject{
+					{
+						Kind:      "ServiceAccount",
+						Name:      controller.ExternalDNSResourceName(testExternalDNS),
+						Namespace: testNamespace,
+					},
+				},
+			},
+			expectChanged: true,
+		},
+		{
+			name: "Subject namespace changed",
+			inputCurrent: &rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: controller.ExternalDNSResourceName(testExternalDNS),
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     controller.ExternalDNSBaseName,
+				},
+				Subjects: []rbacv1.Subject{
+					{
+						Kind:      "ServiceAccount",
+						Name:      "othersa",
+						Namespace: testNamespace,
+					},
+				},
+			},
+			inputExpected: &rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: controller.ExternalDNSResourceName(testExternalDNS),
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     controller.ExternalDNSBaseName,
+				},
+				Subjects: []rbacv1.Subject{
+					{
+						Kind:      "ServiceAccount",
+						Name:      controller.ExternalDNSResourceName(testExternalDNS),
+						Namespace: "otherns",
```

Comment:
I think you want the following:
```suggestion
						Name:      controller.ExternalDNSResourceName(testExternalDNS),
						Namespace: "otherns",
					},
				},
			},
			inputExpected: &rbacv1.ClusterRoleBinding{
				ObjectMeta: metav1.ObjectMeta{
					Name: controller.ExternalDNSResourceName(testExternalDNS),
				},
				RoleRef: rbacv1.RoleRef{
					APIGroup: "rbac.authorization.k8s.io",
					Kind:     "ClusterRole",
					Name:     controller.ExternalDNSBaseName,
				},
				Subjects: []rbacv1.Subject{
					{
						Kind:      "ServiceAccount",
						Name:      controller.ExternalDNSResourceName(testExternalDNS),
						Namespace: testNamespace,
```
That is, expected subject name and namespace and current subject name are the usual values, but current subject namespace is something other.